SwiftDungeon/Player.swift

```diff
@@ -50,49 +66,51 @@ class Player : Entity {
                             SKTexture(imageNamed: "rogue_idle_09"),
                             SKTexture(imageNamed: "rogue_idle_10")]
         
-        let animationAction = SKAction.animate(with: textureAnimation, timePerFrame: 0.2)
+        animationAction = SKAction.animate(with: textureAnimation, timePerFrame: 0.2)
         let repeatAction = SKAction.repeatForever(animationAction)
         
         self.run(repeatAction)
     }
     
     //Attack animation/action
     func attack() {
-        textureAnimation = [SKTexture(imageNamed: "rogue_attack_01"),
-                            SKTexture(imageNamed: "rogue_attack_02"),
-                            SKTexture(imageNamed: "rogue_attack_03"),
-                            SKTexture(imageNamed: "rogue_attack_04"),
-                            SKTexture(imageNamed: "rogue_attack_05"),
-                            SKTexture(imageNamed: "rogue_attack_06"),
-                            SKTexture(imageNamed: "rogue_attack_07"),
-                            SKTexture(imageNamed: "rogue_attack_08"),
-                            SKTexture(imageNamed: "rogue_attack_09"),
-                            SKTexture(imageNamed: "rogue_attack_10"),
-                            SKTexture(imageNamed: "rogue_attack_11")]
-        
-        let animationAction = SKAction.animate(with: textureAnimation, timePerFrame: 0.08)
-        //let repeatAction = SKAction.runAction(animationAction)
-        
-        self.run(animationAction)
+        if (!isAnimating) {
+            isAnimating = true
+            textureAnimation = [SKTexture(imageNamed: "rogue_attack_01"),
```

Comment:
Loading textures each time you attack? Load once, reference as much as you like.